cli/text2embeddings.py

```diff
@@ -213,33 +217,40 @@ def run_embeddings_generation(
                 extra={"props": {"task_output_path": task_output_path, "exception": e}},
             )
 
-        embeddings_output_path = os.path.join(output_dir, task.document_id + ".npy")
+        for encoder_name, encoder in encoders.items():
+            encoder_name_slug = encoder_name_as_slug(encoder_name)
+            embeddings_output_path = os.path.join(
+                output_dir, task.document_id + "__" + encoder_name_slug + ".npy"
+            )
 
-        file_exists = (
-            check_file_exists_in_s3(embeddings_output_path)
-            if s3
-            else os.path.exists(embeddings_output_path)
-        )
-        if file_exists:
-            logger.info(
-                f"Embeddings output file '{embeddings_output_path}' already exists, "
-                "skipping processing."
+            file_exists = (
+                check_file_exists_in_s3(embeddings_output_path)
+                if s3
+                else os.path.exists(embeddings_output_path)
             )
-            continue
+            if file_exists:
+                logger.info(
+                    f"Embeddings output file '{embeddings_output_path}' already exists, "
+                    "skipping processing."
+                )
+                continue
 
-        description_embedding, text_embeddings = encode_parser_output(
-            encoder, task, config.ENCODING_BATCH_SIZE, device=device
-        )
+            logger.info(
+                f"Encoding text using model {encoder_name}.",
+            )
+            description_embedding, text_embeddings = encode_parser_output(
+                encoder, task, config.ENCODING_BATCH_SIZE, device=device
+            )
 
-        combined_embeddings = (
-            np.vstack([description_embedding, text_embeddings])
-            if text_embeddings is not None
-            else description_embedding.reshape(1, -1)
-        )
+            combined_embeddings = (
+                np.vstack([description_embedding, text_embeddings])
+                if text_embeddings is not None
+                else description_embedding.reshape(1, -1)
+            )
```

Comment:
Just thinking about this. It makes sense to combine embeddings for product search to retain document context, however for RAG, it'll be less useful, because we're doing single-document in the first pass. Not for this PR, but one to think about.

Reply:
agree that they're not needed for RAG, but i want to do this work with the minimal possible changes to the indexer so will leave for now